apps/full-site-editing/full-site-editing-plugin/components/post-autocomplete/index.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import { map, throttle } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Popover, Spinner, TextControl } from '@wordpress/components';
+import { withState } from '@wordpress/compose';
+import { select } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const updateSuggestions = throttle( async ( search, setState ) => {
+	setState( {
+		loading: true,
+		showSuggestions: true,
+		suggestions: [],
+	} );
+
+	const suggestions = await apiFetch( {
+		path: addQueryArgs( '/wp/v2/search', {
+			context: 'embed',
+			per_page: 20,
+			search,
+		} ),
+	} );
+
+	setState( {
+		loading: false,
+		showSuggestions: true,
+		suggestions,
+	} );
+} );
+
+const selectSuggestion = async ( suggestion, setState ) => {
+	setState( {
+		loading: true,
+		search: suggestion.title,
+		showSuggestions: false,
+		suggestions: [],
+	} );
+
+	const postType = select( 'core' ).getEntityRecord( 'root', 'postType', suggestion.subtype );
+	const selectedPost = await apiFetch( {
+		path: `/wp/v2/${ postType.rest_base }/${ suggestion.id }`,
+	} );
+
+	setState( {
+		loading: false,
+	} );
+
+	return selectedPost;
+};
+const PostAutocomplete = withState( {
+	loading: false,
+	search: null,
+	showSuggestions: false,
+	suggestions: [],
+} )(
+	( {
+		loading,
+		onSelectPost,
+		search,
+		selectedPostTitle,
+		setState,
+		showSuggestions,
+		suggestions,
+	} ) => {
+		const onChange = async inputValue => {
+			setState( { search: inputValue } );
+			if ( inputValue.length < 2 ) {
+				setState( {
+					loading: false,
+					showSuggestions: false,
+				} );
+				return;
+			}
+			await updateSuggestions( inputValue, setState );
```

Comment:
what's the purpose of the `await` here?

`updateSuggestions()` will return a Promise under the hood but we're neither waiting on the resolution of that promise nor using its returned value, so I think we can disregard the `await` and the `async` in the function signature

Reply:
Good catch.
It was a leftover from a temporary fix long gone. 🙂